SU2_CFD/src/output/filewriter/CParallelDataSorter.cpp

```diff
@@ -213,60 +193,43 @@ void CParallelDataSorter::SortOutputData() {
   /*--- Wait for the non-blocking sends and recvs to complete. ---*/
 
 #ifdef HAVE_MPI
+  MPI_Status status;
+  int ind;
+
   int number = 2*nSends;
   for (int ii = 0; ii < number; ii++)
-    SU2_MPI::Waitany(number, send_req, &ind, &status);
+    MPI_Waitany(number, send_req, &ind, &status);
 
   number = 2*nRecvs;
   for (int ii = 0; ii < number; ii++)
-    SU2_MPI::Waitany(number, recv_req, &ind, &status);
+    MPI_Waitany(number, recv_req, &ind, &status);
 
   delete [] send_req;
   delete [] recv_req;
 #endif
 
-  /*--- Note, passiveDoubleBuffer and doubleBuffer point to the same address.
-   * This is the reason why we have to do the following copy/reordering in two steps. ---*/
+  /*--- Reorder the data in the buffer. ---*/
 
-  /*--- Step 1: Extract the underlying double value --- */
+  vector<passivedouble> tmpBuffer(nPoint_Recv[size]);
 
-  if (!std::is_same<su2double, passivedouble>::value){
-    for (int jj = 0; jj < VARS_PER_POINT*nPoint_Recv[size]; jj++){
-      const passivedouble tmpVal = SU2_TYPE::GetValue(doubleBuffer[jj]);
-      passiveDoubleBuffer[jj] = tmpVal;
-      /*--- For some AD datatypes a call of the destructor is
-       *  necessary to properly delete the AD type ---*/
-      doubleBuffer[jj].~su2double();
-    }
-  }
-
-  /*--- Step 2: Reorder the data in the buffer --- */
```

Comment:
These last changes were to fix issues when the RealReverseIndex is used.
We used to have a char buffer that was used for both passive and active doubles, which required these explicit calls to the destructor, but I guess that did not work so well with mpi and all.
Now there is only passivedouble, which means that once something goes into the data sorter AD information is lost.
This would matter if some computation was performed with time averaged values, which does not seem to be the case at the moment...